receiver/skywalkingreceiver/internal/metrics/metric_report_service.go

```diff
@@ -52,18 +53,21 @@ func NewReceiver(nextConsumer consumer.Metrics, set receiver.CreateSettings) (*R
 
 // Collect implements the service Collect traces func.
 func (r *Receiver) Collect(ctx context.Context, jvmMetricCollection *agent.JVMMetricCollection) (*common.Commands, error) {
-	err := consumeMetrics(ctx, jvmMetricCollection, r.nextConsumer)
+	err := consumeMetrics(ctx, jvmMetricCollection, r.nextConsumer, r.grpcObsrecv)
 	if err != nil {
 		return &common.Commands{}, err
 	}
 	return &common.Commands{}, nil
 }
 
-func consumeMetrics(ctx context.Context, collection *agent.JVMMetricCollection, nextConsumer consumer.Metrics) error {
+func consumeMetrics(ctx context.Context, collection *agent.JVMMetricCollection, nextConsumer consumer.Metrics, obsreport *receiverhelper.ObsReport) error {
 	if collection == nil {
 		return nil
 	}
 	pmd := SwMetricsToMetrics(collection)
-	return nextConsumer.ConsumeMetrics(ctx, pmd)
+	obsContext := obsreport.StartMetricsOp(ctx)
+	err := nextConsumer.ConsumeMetrics(ctx, pmd)
+	obsreport.EndMetricsOp(obsContext, format, pmd.MetricCount(), err)
```

Comment:
From what I have seen before you should store the value of `pmd.MetricCount()` in a temporary variable and then pass that var as a parameter. I believe not doing this could lead to a panic if `pmd` is mutated later on.

Reply:
You are right ,  I have add a patch